pkg/deploy/api/validation/validation.go

```diff
@@ -154,6 +154,14 @@ func validateLifecycleHook(hook *deployapi.LifecycleHook, pod *kapi.PodSpec, fld
 		errs = append(errs, field.Required(fldPath.Child("failurePolicy"), ""))
 	}
 
+	if hook.MaxRetries > 0 && hook.FailurePolicy != deployapi.LifecycleHookFailurePolicyRetry {
+		errs = append(errs, field.Invalid(fldPath, "<hook>", "maxRetries can be used only with the 'Retry' failurePolicy"))
+	}
+
+	if hook.MaxRetries < 0 && hook.FailurePolicy == deployapi.LifecycleHookFailurePolicyRetry {
```

Comment:
@deads2k I don't, validation will refuse them, @kargakis was asking for negative check.